File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 8/20/2026, 2:20:02 AM
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        
        l, r = 1, 1
        while r < len(nums):
            if nums[r] != nums[r - 1]:
                 nums[l] = nums[r]
                 l += 1
            r += 1

        return l 

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 8/20/2026, 2:20:07 AM
class Solution {
    public int removeDuplicates(int[] nums) {
        var i = 0;
        for(int n: nums){
            if(i==0 || n > nums[i - 1]){
                nums[i++] = n;
            }
        }
        return i;
    }
}